plugin.rb

```diff
@@ -160,39 +160,37 @@ class ::User
     post_ids = topic_view.posts.pluck(:id)
     post_ids_sql = post_ids.join(",")
 
-    comment_post_ids_sql = <<~SQL
+    comment_ids_sql = <<~SQL
     SELECT
-      post_replies.reply_post_id
-    FROM post_replies
+      question_answer_comments.id
+    FROM question_answer_comments
     INNER JOIN LATERAL (
       SELECT 1
       FROM (
         SELECT
-          posts.id AS post_id
-        FROM posts
-        INNER JOIN post_replies pr2 ON posts.id = pr2.reply_post_id
-        WHERE pr2.post_id = post_replies.post_id
-        AND posts.post_type = #{Post.types[:regular].to_i}
-        ORDER BY posts.post_number ASC
-        LIMIT 2
+          qa_comments.id
+        FROM question_answer_comments qa_comments
+        WHERE qa_comments.post_id = question_answer_comments.post_id
+        ORDER BY qa_comments.id ASC
+        LIMIT #{TopicView::PRELOAD_COMMENTS_COUNT}
       ) X
-      WHERE X.post_id = post_replies.reply_post_id
+      WHERE X.id = question_answer_comments.id
     ) Y ON true
-    WHERE post_replies.post_id IN (#{post_ids_sql})
+    WHERE question_answer_comments.post_id IN (#{post_ids_sql})
     SQL
 
-    Post.where("id IN (#{comment_post_ids_sql})").order(post_number: :asc).each do |post|
-      topic_view.comments[post.reply_to_post_number] ||= []
-      topic_view.comments[post.reply_to_post_number] << post
+    QuestionAnswerComment.where("id IN (#{comment_ids_sql})").each do |qa_comment|
+      topic_view.comments[qa_comment.post_id] ||= []
+      topic_view.comments[qa_comment.post_id] << qa_comment
     end
 
     comments_counts_sql = <<~SQL
     SELECT
-      post_replies.post_id,
+      question_answer_comments.post_id,
       COUNT(*) AS comments_count
-    FROM post_replies
-    WHERE post_replies.post_id IN (#{post_ids_sql})
-    GROUP BY post_replies.post_id
+    FROM question_answer_comments
+    WHERE question_answer_comments.post_id IN (#{post_ids_sql})
+    GROUP BY question_answer_comments.post_id
```

Comment:
(I have to admit sometimes I wonder when we choose to execute sql statements and when we decide to rely on model scopes.)

Reply:
Good catch! I have no idea why I wrote this in SQL instead of using ActiveRecord.